opengrok-indexer/src/main/java/org/opengrok/indexer/index/IndexDatabase.java

```diff
@@ -495,7 +495,11 @@ public void update() throws IOException {
 
                     IndexDownArgs args = new IndexDownArgs();
                     Statistics elapsed = new Statistics();
-                    LOGGER.log(Level.INFO, "Starting traversal of directory {0}", dir);
+                    if (dir.isEmpty()) {
+                        LOGGER.log(Level.INFO, "Starting traversal of directory 'source root directory'");
+                    } else {
+                        LOGGER.log(Level.INFO, "Starting traversal of directory  \"{0}\" .", dir);
```

Comment:
please avoid escapes, use 2 single quotes as per https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/text/MessageFormat.html 
> Within a String, a pair of single quotes can be used to quote any arbitrary characters except single quotes. For example, pattern string "'{0}'" represents string "{0}", not a FormatElement. A single quote itself must be represented by doubled single quotes '' throughout a String. For example, pattern string "'{''}'" is interpreted as a sequence of '{ (start of quoting and a left curly brace), '' (a single quote), ...

Reply:
@vladak 
LOGGER.log(Level.INFO, "Starting traversal of directory  " ' ' {0}' ' " .", dir);
Are you saying this?

Reply:
almost. just try to compile and run it.